CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.10.2)
+
+# where to find our CMake modules
+set(CMAKE_MODULE_PATH "$ENV{CARMA_OPT_DIR}/cmake")
+# Required for std::lock and syntax
+set(CMAKE_CXX_STANDARD 17)
+
+# Needed so CMake can find the vendored PROJ4 module file. Th FindPROJ4.cmake
+# module file can be removed if we upgrade to a more recent PROJ version. See
+# https://github.com/usdot-fhwa-stol/carma-platform/issues/2139 for updates.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+project(carma-geodetic
+    DESCRIPTION "CARMA geodetic library"
+    HOMEPAGE_URL https://github.com/usdot-fhwa-stol/carma-geodetic-lib
+	VERSION 0.0.1
+    LANGUAGES CXX
+)
+
+find_package(Threads)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
+include(cmake/dependencies.cmake)
+
+add_library(${PROJECT_NAME} SHARED)
+target_link_libraries( ${PROJECT_NAME}
+ ${PROJ4_LIBRARIES}  # Note: Newer versions of PROJ use PROJ::proj
+  GSL  # Note: Newer versions of GSL use Microsoft.GSL::GSL
+  units::units
+)
+target_include_directories(${PROJECT_NAME} 
+    PUBLIC
+        $<INSTALL_INTERFACE:include>
```

Comment:
This can be removed. This is an older convention, and the functionality has been incorporated into the `install(TARGET)` command.

Reply:
Is this true for CMake 3.10.2 or just newer versions?

Reply:
Yes, this is included in CMake 3.10.

Reply:
After looking at the [tutorial](https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20Export%20Configuration.html) I mentioned earlier it still seems that this convention is practiced. Since this accomplishes the same thing either way I think its probably better we stick with the convention for now to be consistent. For the sake of time and consistency I think making changes that are non-functional like this need more of a business justification but I still think its always a good idea to track any things we can potentially improve.

Reply:
That's fine. We can keep it for now and make an issue elsewhere for an eventual migration. A lot of projects still do it this way, so no big deal.